mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -437,6 +439,10 @@ impl S3CrtClientInner {
         let span_body = request_span.clone();
         let span_finish = request_span;
 
+        let endpoint = options.get_endpoint().expect("S3Message always has an endpoint");
```

Comment:
I don't love this — it's possible to reach this code by constructing a `MetaRequestOptions` yourself — but the APIs here don't make that clear right now. I'd like to refactor a bit to make more explicit that we expect this `options` to always have been constructed by one of our own methods, but I'll defer that for now.